addons/l10n_sa_edi/models/account_move_send.py

```diff
@@ -0,0 +1,24 @@
+from odoo import api, models, _
+
+
+class AccountMoveSend(models.AbstractModel):
+    _inherit = 'account.move.send'
+
+    @api.model
+    def _is_sa_edi_applicable(self, move):
+        zatca_document = move.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'sa_zatca' and d.state == 'to_send')
+        return move.country_code == 'SA' and move.move_type in ('out_invoice', 'out_refund') and zatca_document and move.state != 'draft'
+
+    def _get_all_extra_edis(self) -> dict:
+        # EXTENDS 'account'
+        res = super()._get_all_extra_edis()
+        res.update({'sa_edi': {'label': _("Send to Zatca"), 'is_applicable': self._is_sa_edi_applicable}})
```

Comment:
This would need an entry in the pot and a translation in Arabic in the po, I guess.